Travisdouce/multiple remotes

tdouce opened the pull request:
Allow 'client' can create test data in two different 'remotes'

tdouce committed:
Allow multiple home/remotes to be configured

This allows the “client” to:
1. Configure remotes by name
2. Configure multiple remotes

tdouce committed:
Create factories with configured remotes/homes

This allows the user to create factories with multiple remotes

tdouce committed:
Refactors resetting RemoteFactoryGirl configurations

* Removes RemoteFactoryGirl.reset. Instead reset configuration with
RemoteFactoryGirl.remotes_config#reset
* reset configuration when multiple remotes are configured and when
only one unnamed remote is configured

tdouce committed:
Retrieve list of factories when multiple remotes are configured

tdouce commented:
@gschorkopf @iamvery Can you review this PR if you have a few moments? :cherries:

gschorkopf commented on spec/integration/remote_factory_girl_spec.rb:
```diff
@@ -49,6 +49,41 @@
+
+    context 'when configuring multiple remotes' do
+
+      before do
```
This before block is duplicated a few times. Could you break out the configure part into a spec_helper method? Something like:

``` ruby
def configure_remote_factory(host:, port:, end_point:, return_response_as: :as_hash, return_with_root: true)
  RemoteFactoryGirl.configure(:travis) do |config|
    config.home = {:host => host, :port => port, :end_point => end_point }
    config.return_response_as = return_response_as
    config.return_with_root = return_with_root
  end
end

# In your spec!
before do
  configure_remote_factory(host: 'localhost', port: 3000, end_point: '/remote_factory_girl/travis/home')
end
```

tdouce replied:
Yes!. That is a very good idea. :+1:

gschorkopf commented:
Lookin' good! I'm sure Jay will have some more productive things to say as well.

tdouce committed:
Adds helper method configure_remote_factory_girl

@gschorkopf

tdouce committed:
Refactors RemotesConfig

* Simplifies class dramatically
* Removes unnecessary methods #add_remote, #config_for

tdouce committed:
Removes RemoteFactoryGirl.config setter

This method as seriously deranged because it was supplying to arguments
to the setter method. It actually worked, but this method is removed
because it is never used.

tdouce committed:
Bump version to "1.1.0"

* Configure/Create test data in multiple “home”/“remote” applications
* Remove RemoteFactoryGirl.reset. Instead, use
RemoteFactoryGirl.configs_remote.reset
